### PR TITLE
Server, Executor: Bug fixes and a more detailed startup message

### PR DESCRIPTION
**Pull request by AmoebaProtozoa**
### What problem does this PR solve?

- Fixed a bug related to the Executor module
- Fixed a Unit Test case in server/conn_test.go
- Added new TiDB for PostgreSQL welcome message when connecting with psql client

### What is changed

- In types/datum.go, the default datum type is now KindNull instead of KindInterface, which was causing trouble when planer builder is interpreting type of value expression in sql like: `prepare stmt from 'select ? in (?, ?)';`
- In server/conn_test.go, the dispatch simple query unit test now reconcile with the inherent limitation of our protocol transformation, and will be happy even if we can't get the exact same Postgres table description packet.
- In server/conn.go, a welcoming message is added by injecting into the server version information returned to the client. 

### Effects 
- All unit tests within conn_test.go can now be passed
- Unit tests under executor module that was throwing "cannot convert <nil>(type <nil>) to string" can now be passed

### CODE REVIEW

File: types/datum.go [hunk 1]
```diff
@@ -2339,5 +2339,5 @@ func EstimatedMemUsage(array []Datum, numOfRows int) int64 {
 
 // SetInterfaceType 只设置datum的成员k，不要管x
```
Comment (pupillord) [inline types/datum.go:2340]: You can translate the comment, and explain the function

**Comment by RogueJin** [comment]
LGTM

**Review by RogueJin — APPROVED** [review]
LGTM

**Review by pupillord — APPROVED** [review]
LGTM

**Review by hey-hoho — APPROVED** [review]
LGTM

**Pull request merged**